modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/OnboardingPlusFeaturesPage.kt

```diff
@@ -74,18 +63,34 @@ import au.com.shiftyjelly.pocketcasts.localization.R as LR
 private val background = Color(0xFF121212)
 
 @Composable
-fun OnboardingPlusFeaturesPage(
-    onShown: () -> Unit,
+internal fun OnboardingPlusFeaturesPage(
     onUpgradePressed: () -> Unit,
     onNotNowPressed: () -> Unit,
     onBackPressed: () -> Unit,
+    canUpgrade: Boolean
 ) {
-
     val viewModel = hiltViewModel<OnboardingPlusFeaturesViewModel>()
     val state by viewModel.state.collectAsState()
 
-    LaunchedEffect(Unit) { onShown() }
-    BackHandler { onBackPressed() }
+    @Suppress("NAME_SHADOWING")
+    val onUpgradePressed = {
+        viewModel.onUpgradePressed()
+        onUpgradePressed()
+    }
```

Comment:
I'm curious what you think about this pattern. I've always been skeptical of shadowing variable names, but this seemed like a nice way to guarantee that `onUpgradePressed()` cannot be called without also calling `viewModel.onUpgradePressed`.